libnmstate/nm/macvlan.py

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from libnmstate.error import NmstateValueError
+from libnmstate.schema import MacVlan
+from .common import NM
+
+
+MODE_TO_NM_NUMBER = {
+    MacVlan.Mode.UNKNOWN: 0,
+    MacVlan.Mode.VEPA: 1,
+    MacVlan.Mode.BRIDGE: 2,
+    MacVlan.Mode.PRIVATE: 3,
+    MacVlan.Mode.PASSTHRU: 4,
+    MacVlan.Mode.SOURCE: 5,
+}
+
+
+def create_setting(iface_state, base_con_profile):
+    macvlan = iface_state.get(MacVlan.TYPE)
+    if not macvlan:
+        return None
```

Comment:
How about moving these check to `nm.profile`.

Reply:
I've been thinking about this. In order to do this it would mean we need to import all the schema on nm/profile.py and as we discussed nm/profile.py should handle profile related code. IMO, this is only related to MacVlan support. Anyway, as this change would be needed for all the interfaces, I will propose it on a different PR so this way we can discuss it properly. What do you think?

Reply:
I am OK with either way. Not a big deal.